### PR TITLE
Support serialization and deserialization for enumerable/lists of OneOf Cases in class fields.

### PR DESCRIPTION
**Pull request by r-arjocu**
Fixes this issue: https://github.com/dpraimeyuu/OneOf.Serialization/issues/1

### CODE REVIEW

File: OneOf.Serialization.Tests/OneOfJsonConverterTests.cs [hunk 1]
```diff
@@ -21,6 +23,43 @@ public void Given_Instance_Of_OneOf_Case_When_Serializing_Then_Produces_Valid_Js
             var json = JsonConvert.SerializeObject(engine, settings);
             var engineJObject = JObject.FromObject(engine);
             json.Should().NotBeNullOrEmpty();
+            json.Should().StartWith("{").And.EndWith("}");
```
Comment (dpraimeyuu) [inline OneOf.Serialization.Tests/OneOfJsonConverterTests.cs:26]: @ra-mt what is the intention behind that? To verify we are dealing with an object?

File: OneOf.Serialization.Tests/OneOfJsonConverterTests.cs [hunk 2]
```diff
@@ -21,6 +23,43 @@ public void Given_Instance_Of_OneOf_Case_When_Serializing_Then_Produces_Valid_Js
...
+            var engines = new List<Engine> ()
+            {
+                new WorkingEngine(Power.Create(999)),
+                new WorkingEngine(Power.Create(888)),
+            };
+            engines[1].AsT0.Status = new Status.Started(20);
+            var json = JsonConvert.SerializeObject(engines.AsEnumerable(), settings);
+            var enginesJArray = JArray.FromObject(engines.AsEnumerable());
+            json.Should().NotBeNullOrEmpty();
+            json.Should().StartWith("[").And.EndWith("]");
```
Comment (dpraimeyuu) [inline OneOf.Serialization.Tests/OneOfJsonConverterTests.cs:42]: @ra-mt what is the intention behind that? If we are dealing with JSON array?

File: OneOf.Serialization.Tests/OneOfJsonConverterTests.cs [hunk 3]
```diff
@@ -44,13 +83,61 @@ public void Given_Json_With_OneOf_Case_When_Deserializing_Then_Produces_Correct_
...
+            engineEnumerable.Should().NotBeNull();
+
+            var engineList = engineEnumerable.ToList();
+
+            var engine0 = engineList[0];
+            engine0.Should().NotBeNull();
+            engine0.Value.Should().BeOfType<WorkingEngine>();
+            var engine0Status = (engine0.Value as WorkingEngine).Status;
+            engine0Status.Should().NotBeNull();
+            engine0Status.Value.Should().BeOfType<Status.Idle>();
```
Comment (dpraimeyuu) [inline OneOf.Serialization.Tests/OneOfJsonConverterTests.cs:107]: Does it make sense to create a tiny extension method hiding the details and expressing something like:
```cs
workingEngine0.ShouldBeIdle();
```

:thinking:

**Review by dpraimeyuu — APPROVED** [review]
LGTM, awesome work, @ra-mt :rocket:

File: OneOf.Serialization.Tests/OneOfJsonConverterTests.cs (hunk 1 above)
Comment (r-arjocu) [reply OneOf.Serialization.Tests/OneOfJsonConverterTests.cs:26]: Yes, to verify that it's a JSON object.

File: OneOf.Serialization.Tests/OneOfJsonConverterTests.cs (hunk 2 above)
Comment (r-arjocu) [reply OneOf.Serialization.Tests/OneOfJsonConverterTests.cs:42]: Yes, to verify that we get a JSON array.

File: OneOf.Serialization.Tests/OneOfJsonConverterTests.cs (hunk 3 above)
Comment (r-arjocu) [reply OneOf.Serialization.Tests/OneOfJsonConverterTests.cs:107]: That's more up to you. I've only copied that code. But for symmetry, a similar method `workingEngine1.ShouldBeStarted(readiness)` should be created.

Making similar extension methods would be a bit more difficult for my next pull request though, which contains significantly more code.

**Comment by r-arjocu** [comment]
Unfortunately I don't see any "Merge" button. I think only you can merge.

**Comment by dpraimeyuu** [comment]
@ra-mt merged, thanks for your contribution!

**Comment by r-arjocu** [comment]
> @ra-mt merged, thanks for your contribution!

You're welcome!

**Pull request merged**